.github/workflows/on-prerelease.yml

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+name: On Pre-Release
+on:
+  release:
+    types:
+      - prereleased
+
+env:
+  ARTIFACTS_API_KEY: ${{ secrets.AZURE_ARTIFACTS_PERSONAL_ACCESS_TOKEN }}
+  GITHUB_TOKEN: ${{ secrets.PAT_ATTACH_TO_RELEASE }}
+
+defaults:
+  run:
+    working-directory: Meadowlark-js
+
+jobs:
+  azure-publish:
+    name: Publish to Azure Artifacts
+    runs-on: ubuntu-latest
+    outputs:
+      hash-code: ${{ steps.hash-code.outputs.hash-code }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          # Need full Git history for calculating the version number
+          fetch-depth: 0
+
+      - name: Setup Node
+        uses: actions/setup-node@8c91899e586c5b171469028077307d293428b516 # v3.5.1
+        with:
+          node-version: "16"
+          cache: "npm"
+          cache-dependency-path: "**/package-lock.json"
+
+      - name: Install dependencies
+        if: ${{ steps.modules-cache.outputs.cache-hit != 'true' }}
+        run: npm install
+
+      - name: Update version to match tag
+        working-directory: eng/version
+        run: |
+          npm install
+          node ./index.js --update
+
+      - name: Transpile to JS
+        run: npm run build --workspaces
+
+      - name: Create .npmrc for Azure Artifacts publishing
+        run: |
+          set +x  # disable echo to shell so that the base64-encoded API key never escapes into the log.
+
+          # Create the user .npmrc file with proper settings
+          API_KEY=`echo -n "${{ env.ARTIFACTS_API_KEY }}" | base64`
+          touch $HOME/.npmrc
+          echo "; begin auth token" >> $HOME/.npmrc
+          echo //pkgs.dev.azure.com/ed-fi-alliance/Ed-Fi-Alliance-OSS/_packaging/EdFi/npm/registry/:username=ed-fi-alliance >> $HOME/.npmrc
+          echo //pkgs.dev.azure.com/ed-fi-alliance/Ed-Fi-Alliance-OSS/_packaging/EdFi/npm/registry/:_password=$API_KEY >> $HOME/.npmrc
+          echo //pkgs.dev.azure.com/ed-fi-alliance/Ed-Fi-Alliance-OSS/_packaging/EdFi/npm/registry/:email=techsupport@ed-fi.org >> $HOME/.npmrc
+          echo "; end auth token" >> $HOME/.npmrc
+
+      - name: Publish
+        run: npm publish --workspaces --tag dev
+
+      # NOTE: there are uncommitted version number changes in package.json. No
+      # problem! Our strategy is to rely on the tag history for version numbers,
+      # not on package.json.
```

Comment:
@bradbanister this could end up coming back to haunt me.

In MetaEd, we have a GPG key for signing a commit and tag, which we can then push back to the repository. I wasn't thrilled about that process, but I accepted it. In this public repository I am even less enthusiastic about it, as there is more opportunity for a secret to spill out to the public... giving someone who manages to steal the GPG key full access to write commits into our repositories. Hence I'm hoping this version update will work even though it isn't being preserved in the package.json files.

Reply:
Yes, I think the package.json versions getting out of sync will come back to haunt us.

It looks like in the past 6 months GitHub has added a fine-grained permissions option to access tokens: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#fine-grained-personal-access-tokens

Maybe we could use that to limit access to commit pushes in Meadowlark?